lib/wallaroo/core/source/connector_source/connector_framed_source_notify.pony

```diff
@@ -16,24 +16,155 @@ Copyright 2017-2019 The Wallaroo Authors.
+    last_seen': PointOfReference,
+    last_acked': PointOfReference, last_checkpoint': CheckpointId)
+=>
+  id = stream_id
+  name = stream_name
+  last_seen = last_seen'
+  last_acked = last_acked'
+  last_checkpoint = last_checkpoint'
+
+  fun serialize(wb: Writer = Writer): Writer =>
+    wb.u64_be(id)
+    wb.u16_be(name.size().u16())
+    wb.write(name)
+    wb.u64_be(last_acked)
+    wb.u64_be(last_seen)
+    wb.u64_be(last_checkpoint)
+    wb
+
+  new ref deserialize(rb: Reader) ? =>
+    id = rb.u64_be()?
+    let name_size = rb.u16_be()?.usize()
+    name = String.from_array(rb.block(name_size)?)
+    last_acked = rb.u64_be()?
+    last_seen = rb.u64_be()?
+    last_checkpoint = rb.u64_be()?
+
+class val NotifyResult[In: Any val]
+  """
+  The type to use with Promise objects used to manage responses for the async
+  parts of the stream_notify sequence
+  """
+  let source: ConnectorSource[In] tag
+  let success: Bool
+  let stream: StreamTuple
+
+  new val create(source': ConnectorSource[In] tag,
+    success': Bool, stream': StreamTuple)
+  =>
+    source = source'
+    success = success'
+    stream = stream'
 
+class NotifyResultFulfill[In: Any val] is
+  Fulfill[NotifyResult[In], None]
+
+  let stream_id: StreamId
+  let session_id: RoutingId
+
+  new create(stream_id': StreamId, session_id': RoutingId)
+  =>
+    stream_id = stream_id'
+    session_id = session_id'
+
+  fun apply(t: NotifyResult[In]) =>
+    t.source.stream_notify_result(session_id, t.success, t.stream)
+
+class val ConnectorSourceNotifyParameters[In: Any val]
+  let pipeline_name: String
+  let env: Env
+  let auth: AmbientAuth
+  let handler: FramedSourceHandler[In] val
+  let runner_builder: RunnerBuilder
+  let partitioner_builder: PartitionerBuilder
+  let router: Router
+  let metrics_reporter: MetricsReporter val
+  let event_log: EventLog
+  let target_router: Router
+  let cookie: String
+  let max_credits: U32
+  let refill_credits: U32
+  let host: String
+  let service: String
+
+  new val create(pipeline_name': String, env': Env,
+    auth': AmbientAuth, handler': FramedSourceHandler[In] val,
+    runner_builder': RunnerBuilder, partitioner_builder': PartitionerBuilder,
+    router': Router, metrics_reporter': MetricsReporter val,
+    event_log': EventLog, target_router': Router, cookie': String,
+    max_credits': U32, refill_credits': U32, host': String, service': String)
+  =>
+    pipeline_name = pipeline_name'
+    env = env'
+    auth = auth'
+    handler = handler'
+    runner_builder = runner_builder'
+    partitioner_builder = partitioner_builder'
+    router = router'
+    metrics_reporter = metrics_reporter'
+    event_log = event_log'
+    target_router = target_router'
+    cookie = cookie'
+    max_credits = max_credits'
+    refill_credits = refill_credits'
+    host = host'
+    service = service'
 
 class ConnectorSourceNotify[In: Any val]
-  let _source_id: RoutingId
+  let source_id: RoutingId
```

Comment:
Why this change?  Shouldn't the `ConnectorSource` already know the source id, and isn't the `ConnectorSource` the only thing with a reference to this Notify?